src/transaction/atomic_replication_helper.cpp

```diff
@@ -889,6 +888,13 @@ namespace cublog
     // sysop end parent lsa can also be null
   }
 
+  atomic_replication_helper::atomic_log_sequence::atomic_log_entry::~atomic_log_entry ()
+  {
+    // this assert should be valid; except we're using emplaced constructed objects for which
+    // we're not properly initializing this value
+    //assert (m_page_ptr == nullptr);
```

Comment:
```suggestion
    assert (m_page_ptr == nullptr);
```

Reply:
Or intended?

Reply:
No, it is intended. The comment explains the cause. However I have removed the dtor altogether as it became superfluous.